### ISSUE TITLE
Main process keeps on running even when a Thread is down.

### ISSUE THREAD

**Issue by kinvaris** [psychomario/PyPXE#138]
Problem: When a thread goes down, the main process keeps on running.
Situated: https://github.com/psychomario/PyPXE/blob/master/pypxe/server.py#L289

The while just checks if there is a map available. It does not check the availability of the running services.

Seen here:

```
In [6]: bool([True, True, False])
Out[6]: True

In [7]: bool([True, True, True])
Out[7]: True

In [8]: bool([])
Out[8]: False
```

I don't know if this is unintended or just to check if there are any services added to the list?

**Comment by psychomario** [comment]
Good catch. The intention was to do something like
```
filter(None, map(lambda x: x.isAlive(), running_services))
```
But apparently it was never done. This still doesn't implement any state for if a service crashes.

**Comment by kinvaris** [comment]
You could fix it like this:
```
In [1]: x = [True, True, True]

In [2]: all(el==True for el in x)
Out[2]: True

In [3]: x = [True, True, False]

In [4]: all(el==True for el in x)
Out[4]: False
```

**Comment by mmattioli** [comment]
@kinvaris if you want to submit a patch to `development` you're more than welcome 😄

**Comment by psychomario** [comment]
`==True` is implicit I believe, so you can just do
```
>>> x = [True, True, True]
>>> all(x)
True
>>> x = [True, True, False]
>>> all(x)
False
```

As @mmattioli said, a patch would be welcomed, otherwise it might be a while until I can get around to this.

**Comment by psychomario** [comment]
Fixed in development, #142
